io/disk/Avago_storage_adapter/avago9361_vd.py

```diff
@@ -150,6 +156,99 @@ def test_online_offline(self):
                 time.sleep(10)
         self.vd_delete()
 
+    def test_ghs_dhs(self):
+
+        """
+        Test to create Global hot spare and Dedicated hot spare
+        """
+        self.vd_create('WT', 'nora', 'direct', 512)
+        cmd = "./storcli64 /c%d/%s add hotsparedrive" % (self.controller,
+                                                         self.hotspare)
+        self.check_pass(cmd, "Failed to create GHS")
+        self.set_online_offline('offline')
+        self.rebuild_progress(self.hotspare)
+        self.copyback_operation()
+        cmd = "./storcli64 /c%d/%s delete hotsparedrive" % (self.controller,
+                                                            self.hotspare)
+        self.check_pass(cmd, "Failed to delete hotsparedrive")
+        cmd = "./storcli64 /c%d/%s add hotsparedrive DGs=0" % (self.controller,
+                                                               self.hotspare)
+        self.check_pass(cmd, "Failed to set DHS")
+        self.set_online_offline('offline')
+        self.rebuild_progress(self.hotspare)
+        self.copyback_operation()
+        self.vd_delete()
+
+    def copyback_state(self, state):
+
+        """
+        Helper function to run copyback test
+        """
+        if state == 'start':
+            cmd = "./storcli64 /c%d/%s start copyback target=%s" \
+                   % (self.controller, self.hotspare, self.copyback)
+        else:
+            cmd = "./storcli64 /c%d/%s %s copyback" % (self.controller,
+                                                       self.on_off, state)
+        self.check_pass(cmd, "Failed to start copyback")
+
+    def copyback_operation(self):
+
+        """
+        Function to run copyback operations
+        """
+        for state in ['start', 'stop', 'start', 'pause', 'resume']:
+            self.copyback_state(state)
+            cmd = "./storcli64 /c%d/%s show copyback" % (self.controller,
+                                                         self.on_off)
+            self.showprogress(cmd)
```

Comment:
I have used each command only once, and there is a anyway helper function for show progress. And moreover, we need to pass VD number for few operations, and for others we need to pass disk slot and also few command expects in different formats. So, its better to be in this way.